arch/arm/core/aarch64/mmu/arm_mmu.c

```diff
@@ -185,14 +95,30 @@ static void set_pte_block_desc(uint64_t *pte, uint64_t addr_pa,
 {
 	uint64_t desc = addr_pa;
 	unsigned int mem_type;
+	unsigned int rwx_el0, rwx_elh;
+
+	rwx_el0 = GET_RWX_EL0(attrs);
+	rwx_elh = GET_RWX_ELh(attrs);
 
 	desc |= (level == 3) ? PTE_PAGE_DESC : PTE_BLOCK_DESC;
 
 	/* NS bit for security memory access from secure state */
 	desc |= (attrs & MT_NS) ? PTE_BLOCK_DESC_NS : 0;
 
-	/* AP bits for Data access permission */
-	desc |= (attrs & MT_RW) ? PTE_BLOCK_DESC_AP_RW : PTE_BLOCK_DESC_AP_RO;
+	/*
+	 * AP bits for EL0 / ELh Data access permission
+	 *
+	 *   AP[2:1]   ELh  EL0
+	 * +--------------------+
+	 *     00      RW   NA
```

Comment:
Just confirming, NA is "No RW" permission?

Reply:
NA is No Access